component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/validator/JWTValidator.java

```diff
@@ -583,6 +612,7 @@ private boolean validateSignature(SignedJWT signedJWT, X509Certificate x509Certi
                 PublicKey publicKey = x509Certificate.getPublicKey();
                 if (publicKey instanceof RSAPublicKey) {
                     verifier = new RSASSAVerifier((RSAPublicKey) publicKey);
+                    verifier.getJCAContext().setProvider(BouncyCastleProviderSingleton.getInstance());
```

Comment:
Why is this required?

Reply:
Had to use this as the PS256 algorithm was not supported and therefore got issues when executing the unit tests. But this is not required when the actual code is being executed.

Reply:
Then can we remove this and figure out another way for unit tests

Reply:
I did try out different ways to keep the PS256 algorithm in the unit tests but it seems there is no way out of it other than using BouncyCastleProviderSingleton. Yet as the requirement is to check the logic within the methods irrespective of the algorithm being used (as it can change time to time based on the spec updates), I changed the test cases assuming that RS512 is a FAPI allowed signature algorithm. Hope that is fine for the tests. The changes are in add0813c8207f09dc2c3c2e0bb9e4e9d3d28f12a